mirascope/prompts.py

```diff
@@ -26,6 +32,36 @@ def _format_template(prompt: Prompt, template: str) -> str:
     return template.format(**{var: getattr(prompt, var) for var in template_vars})
 
 
+class CallParams(BaseModel):
```

Comment:
does this limit us when we implement other models. feels like prompt should take in a union of different call params and the first one we support is OpenAICallParams

Reply:
Good call. Updated